### ISSUE TITLE
RSS feed for playlist

### ISSUE THREAD

**Issue by reesericci** [Chocobozzz/PeerTube#4135]
**Describe the problem to be solved**
I would like to be able to play a playlist via my podcast player. Or get the contents of a playlist via json.

**Describe the solution you would like:**

Provide an RSS/JSON feed (like you do for channels) for playlists.

**Comment by frankstrater** [comment]
Interesting idea. The output could be similar to what YT does;

https://www.youtube.com/feeds/videos.xml?playlist_id=PLzWr6PnXjYMv2xdYzQm8tDg0k14HFysJv

and the data is already available in JSON;

https://peertube.cpy.re/api/v1/video-playlists/42/videos

Do you want to watch videos in your podcast player? Or is it for the audio/video (video is audio + image) uploads? Does your podcast player accept mp4 as media (probably will)?

**Comment by reesericci** [comment]
No, I just want to be able to "subscribe" in a reader to just a playlist, not an entire channel. Take the channel XYZ, XYZ makes a series called ZYX. I like the series ZYX but don't like the rest of XYZ's channel. I also happen to use RSS, so I want to be able to "subscribe" to just that series.

**Comment by frankstrater** [comment]
I was confused because you said "podcast player", which suggested you wanted to listen to PeerTube content, but I see now that you mainly want to get notified through RSS when new content gets added to an existing playlist.

**Comment by agates** [comment]
FYI for it's worth, in the podcast RSS namespace (used by the podcast RSS feeds implemented in peertube), we have a type for [playlists](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#list-mediums).  It would depend on implementing the `<podcast:guid>` tag in the RSS feed, but it would be pretty simple.

**Comment by okaziya** [comment]
## Detailed use case and specification

We would like a user's PeerTube deployment to be fully compatible with podcast distribution to Apple Podcasts, Spotify, and other directories. Currently, PeerTube only generates podcast RSS feeds at the **channel level** (`/feeds/podcast/videos.xml?videoChannelId=X`). This means all shows are merged into a single feed, which might be good for a creator who produces a single show. For supporting multiple shows, Playlists are the natural solution. Playlists have titles, descriptions, and artwork — and public playlists are already required to be associated with a channel (`videoChannelId`), so all the show-level metadata needed for podcast directories is already available. If PeerTube generated podcast-compliant RSS feeds per playlist, videos added to the playlist could be consumed as Episodes within a Show.

Could we please add a **podcast RSS feed endpoint for playlists** at `/feeds/podcast/videos.xml?playlistId=XXXX`?

> **Implementation note:** The current podcast feed handler (`generateVideoPodcastFeed` in `video-podcast-feeds.ts`) is tightly coupled to `videoChannel` — it reads from `res.locals.videoChannel` and calls channel-specific helpers like `VideoModel.guessLanguageOrCategoryOfChannel` and `VideoModel.channelHasNSFWContent`. Adding playlist support would likely require a separate handler or significant refactoring, rather than just accepting a new query parameter on the existing route.

### Requirements

#### 1. Full podcast namespace support (same as channel feeds)

The playlist feed should include the same iTunes and Podcasting 2.0 namespace tags that the existing channel feed already generates. Show-level metadata can be resolved via the playlist's associated channel:

- `<itunes:author>` — from the playlist's channel name
- `<itunes:owner>` with `<itunes:email>` — same logic as channel feeds (user must opt in to show email)
- `<itunes:image>` — **from the playlist's thumbnail** (see image requirements below), with fallback to the channel's avatar
- `<itunes:category>` — inherited from the channel, or a new playlist-level field
- `<itunes:explicit>` — inherit from channel or allow per-playlist override
- `<itunes:type>` — `episodic` (default) or `serial`, ideally configurable per playlist
- `<podcast:medium>` — `video`, `audio`, or potentially [`podcastL`](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#list-mediums) (the podcast namespace's list medium designed for curated collections, as @agates [noted above](https://github.com/Chocobozzz/PeerTube/issues/4135#issuecomment-2673377094))
- `<podcast:guid>` — unique per playlist
- All per-episode (`<item>`) tags: `<enclosure>`, `<itunes:duration>`, `<itunes:image>`, `<podcast:transcript>`, `<podcast:alternateEnclosure>`, `<podcast:socialInteract>`, etc.

#### 2. Playlist metadata as show metadata

The playlist's existing fields should map to podcast show metadata:

| Playlist field | Podcast RSS tag |
|---|---|
| Playlist name | `<title>` |
| Playlist description | `<description>` |
| Playlist thumbnail | `<itunes:image>` |
| Playlist channel | `<itunes:author>`, `<podcast:person>` |

#### 3. Image compliance (relates to #7390)

For the feed to be accepted by Apple Podcasts, the playlist thumbnail used for `<itunes:image>` must meet podcast spec requirements:

- Minimum 1400×1400 pixels, recommended 3000×3000
- JPEG or PNG, RGB color space
- Square aspect ratio

This likely means PeerTube should allow uploading higher-resolution playlist thumbnails than the current default. Channel avatar sizing was addressed in #5908 (PeerTube now generates 600×600 and 1500×1500 avatars since [`109e93c`](https://github.com/Chocobozzz/PeerTube/commit/109e93c139d87fb73d7bd9597852ac5f856b8018)), but #7390 covers remaining gaps for podcast feeds (instance logo size, rectangular video thumbnails, and playlist thumbnails).

#### 4. Feed discoverability

- Add `<link rel="alternate" type="application/rss+xml">` in the HTML `<head>` of playlist pages pointing to the podcast feed
- Add the podcast feed URL to the playlist page UI (similar to how channel pages show RSS links)

### Example use case – Content segmentation

PeerTube creators can produce multiple shows within a single channel. For example, a channel like Blender can post news updates, release notes, tutorials, et cetera ([example](https://video.blender.org/c/blender_channel/videos?s=1)), and with the help of playlist-level podcast feeds, the creators can:

1. Create a number of playlists that represent podcastable "Shows"
2. Submit each show as a **separate podcast** to Apple Podcasts, Spotify, etc.
3. Group them under an **Apple Podcasts Channel** (Apple's concept for bundling related shows)
4. Still maintain a **single PeerTube channel** for web viewers who want to browse all content in one place

----

Related issues: #7390, #3761 (Podcasting 2.0 support — implemented), #5908 (bigger avatars — implemented)

**Comment by agates** [comment]
I would not recommend having a single podcast RSS feed for an entire instance.  Rather, you could have a "publisher" feed that links to all of the other channel feeds via remoteItem.

This helps with channel ownership issues as well as keeps a single feed from becoming astronomically large.  Also helps with discovery (what channels does this publisher have?).

https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/examples/publishers/publishers.md
https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/tags/publisher.md

**Comment by Chocobozzz** [comment]
> I would not recommend having a single podcast RSS feed for an entire instance

I'm not sure to see in this thread where this has been suggested? As I understand it, we would simply publish a podcast feed for each playlist.

**Comment by agates** [comment]
> > I would not recommend having a single podcast RSS feed for an entire instance
> 
> I'm not sure to see in this thread where this has been suggested? As I understand it, we would simply publish a podcast feed for each playlist.

Hmm, now that I read back again, you're right.  I must have misinterpreted something on my first read.

However, publisher feeds are still great!  Could be done at the instance or user level.  And can fulfill a specific role where playlists are overkill.

Apologies for the sidetrack.

**Comment by Chocobozzz** [comment]
This has been implemented in 8.2: https://github.com/Chocobozzz/PeerTube/releases/tag/v8.2.0

**Comment by beedell-roke** [comment]
@Chocobozzz, is the RSS feed IETF Atom-adherent? [^1]

[^1]: [`news.ycombinator.com/item?id=26169162`](https://news.ycombinator.com/item?id=26169162)
